processor/src/main/java/org/mapstruct/ap/MappingProcessor.java

```diff
@@ -427,21 +425,22 @@ private <P, R> R process(ProcessorContext context, ModelElementProcessor<P, R> p
         return processors;
     }
 
-    private TypeElement asTypeElement(Element element) {
-        return element.accept(
-            new ElementKindVisitor6<TypeElement, Void>() {
-                @Override
-                public TypeElement visitTypeAsInterface(TypeElement e, Void p) {
-                    return e;
-                }
-
-                @Override
-                public TypeElement visitTypeAsClass(TypeElement e, Void p) {
-                    return e;
-                }
-
-            }, null
-        );
+    private XElement asTypeElement(XElement element) {
+        return element;
```

Comment:
there does not seem to be a way to convert an XElement to an XElementType, but also did not see why this was needed